include/nlohmann/json.hpp

```diff
@@ -2685,7 +2685,7 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         return ptr.contains(this);
     }
 
-    template<typename BasicJsonType>
+    template<typename BasicJsonType, detail::enable_if_t<detail::is_basic_json<BasicJsonType>::value, int> = 0>
```

Comment:
This constraint is incomplete. (`BasicJsonType::string_t`)

Reply:
I took another look. The functions called by these overloads are constrained, so I'm not too worried about people passing in nonsensical `json_pointer` instances.